### ISSUE TITLE
connectionreg: Suggestions/clarifications

### ISSUE THREAD

**Issue by DanielOaks** [ircdocs/modern-irc#32]
Some extra clarifications from @JonasT to look at

1. Clarification of nothing else may be sent before login sequence is completed:

"The server MUST then respond as though the client sent it the MOTD command, i.e. it must send either the successful Message of the Day numerics or the ERR_NOMOTD numeric." -> "The server MUST then respond as though the client sent it the MOTD command, i.e. it must send either the successful Message of the Day numerics or the ERR_NOMOTD numeric. The server MUST NOT send any JOIN, PRIVMSG, MODE or any other not login-related event before this entire login sequence is complete and everything up to the MOTD or ERR_NOMOTD has been sent."

2. Maybe a note should be added for future numerics (since this discussion also was about some future flexibility for new implementation additions and protocol updates), mainly as a heads up to client devs to not make things explode:

"... and at least one RPL_ISUPPORT (005) numeric to the client. The server SHOULD then respond [...]" -> "... and at least one RPL_ISUPPORT (005) numeric to the client. Now, any additional numerics for additional login and server information MAY be sent if supported by this specific server only, although it is not recommended to server authors to send any such additional non-standard numerics unless they were enabled by a related CAP request of the client. The server SHOULD then respond [...]"

**Comment by JonasT** [comment]
I guess technically both are already *kinda* covered by "Until registration is complete, only a limited subset of commands may be accepted by the server." and "The server MAY send other numerics and messages. " which I might have missed before.

However, I would definitely recommend replacing "The server MAY send other numerics and messages." with something like "The server MAY send other numerics and messages, although it is not recommended for server authors to send any additional non-standard numerics unless they were enabled by a related CAP request of the client." just to be super clear that weird additional numerics made up by a specific server should be technically accepted by the client, but that doing such is not recommended unless it is properly introduced with a CAP.

Also it would be nice to expand "Until registration is complete, only a limited subset of commands may be accepted by the server." in a way that also covers the back direction (server must also not send any unusual events before login is complete). I guess that might be possible with less text than my suggestion above though :smile:

**Comment by janicez** [comment]
Only a limited subset of commands SHOULD

**Comment by DanielOaks** [comment]
I expanded the limited subset of commands" line a bit and I'll probably note which commands are typically accepted pre-reg in the text for those commands themselves. As well, I changed it to SHOULD as suggested. I don't think we really need to go more in-depth than that -- the suggested language above is clunky and I wouldn't really want it in a spec unless it can be proved to be correct (and even then I'd want it rewritten and edited to be less clunky).

I'll close this for now, thanks for the suggestions though!
